### ISSUE TITLE
Wrong ownership of authorized_keys causes ssh passwordless login to fail

### ISSUE THREAD

**Issue by jbemmel** [srl-labs/containerlab#757]
clab bind-mounts a single authorized_keys file in 3 different places:
```
[admin@srl1 debug]$ sudo ls -Aln /home/admin/.ssh/authorized_keys
-rw-r--r-- 1 1000 1000 1 Feb  1 21:18 /home/admin/.ssh/authorized_keys
[admin@srl1 debug]$ sudo ls -Al /home/linuxadmin/.ssh/authorized_keys
-rw-r--r-- 1 user user 1 Feb  1 21:18 /home/linuxadmin/.ssh/authorized_keys
[admin@srl1 debug]$ sudo ls -Al /root/.ssh/authorized_keys
-rw-r--r-- 1 user user 1 Feb  1 21:18 /root/.ssh/authorized_keys
```

The file ownership is determined by the user id on the Docker host, and Linux refuses to read it if the ownership does not match the user id inside the container.

Moreover, SRL assumes the filesystem is writable so it can change the ownership; however, during boot there are errors (due to the file being mounted read-only)
```
setfacl: /root/.ssh/authorized_keys: Read-only file system
touch: cannot touch '/root/.ssh/authorized_keys': Read-only file system
chmod: changing permissions of '/root/.ssh/authorized_keys': Read-only file system
```

**Comment by hellt** [comment]
Which user you have configured on your docker host?

**Comment by hellt** [comment]
@jbemmel do you mind checking this build of containerlab that you can obtain by running

```
# this will download the containerlab binary to your PWD
docker run --rm -v $(pwd):/workspace ghcr.io/deislabs/oras:v0.11.1 pull ttl.sh/clab-e2284709:1d
```

**Comment by hellt** [comment]
with a small fix added in #767 I think the feature works as expected
if you run containerlab as a non root user, you will have to use `sudo`; this will make the authorized_keys file to be owned by `root` user with 644 permissions
Since `all` is allowed to read this file, both admin and linuxadmin users will be able to read this file.

I don't know how you made the authorized_keys file be owned by uid 1000...

PS. the reason those files are mounted in `ro` mode is to prevent the srlinux script to modify permissions on it.
PS2. I agree, the better solution would be to create three autz_key files with the same content. I am just lazy
